chainstate/src/detail/mod.rs

```diff
@@ -307,6 +317,95 @@ impl<S: BlockchainStorage> Chainstate<S> {
         Ok(block)
     }
 
+    pub fn get_token_detail(
+        &self,
+        token_id: TokenId,
+    ) -> Result<(Id<Block>, TokenIssuanceTransaction), PropertyQueryError> {
+        let chainstate_ref = self.make_db_tx_ro();
+
+        // Find issuance transaction id by token_id
+        let creation_tx =
+            chainstate_ref
+                .get_token_info(&token_id)?
+                .ok_or(PropertyQueryError::TokensError(
+                    TokensError::TokensNotRegistered(token_id),
+                ))?;
+        let tx_index = chainstate_ref
+            .get_mainchain_tx_index(&OutPointSourceId::Transaction(creation_tx.get_id()))?
```

Comment:
Even though we talked about this and we mentioned that not every user will necessarily have the transaction index, and that's why we added a database entry for issuance tx id vs token id, yet we have this still sitting here.

Don't worry. I'll fix it for you. That's what I'm here for.